### PR TITLE
use puppetlabs-hocon for product.conf

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by mmoll** [comment]
This is the first small step. Once this is in, other HOCON files can get converted - except maybe auth.conf, because there's puppetlabs-puppet_authorization.

File: manifests/server/puppetserver.pp [hunk 1]
```diff
@@ -238,14 +238,23 @@
...
+
   if versioncmp($server_puppetserver_version, '2.7') >= 0 {
     $product_conf_ensure = file
+
+    hocon_setting { 'server_check_for_updates':
+      ensure  => present,
+      path    => $product_conf,
+      setting => 'product.check-for-updates',
+      value   => $server_check_for_updates,
+      require => File[$product_conf],
```
Comment (ekohl) [inline manifests/server/puppetserver.pp]: I'm surprised this isn't autorequired.

Comment (mmoll) [inline manifests/server/puppetserver.pp]: tbh, I didn't try it without.

Comment (ekohl) [inline manifests/server/puppetserver.pp]: I checked the code and it's not there so I'm pretty sure it doesn't :)

**Pull request merged**